geth/account/accounts_test.go

```diff
@@ -109,3 +111,78 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
 	_, err = acctManager.VerifyAccountPassword(keyStoreDir, address.Hex(), TestConfig.Account3.Password)
 	require.NoError(t, err)
 }
+
+// newTestAccManager returns the account manager and the mock node manager
+// that's being used within it for testing purposes.
+func newTestAccManager(t *testing.T) (accManager *account.Manager, nodeManager *common.MockNodeManager) {
+	ctrl := gomock.NewController(t)
+	nodeManager = common.NewMockNodeManager(ctrl)
+	accManager = account.NewManager(nodeManager)
+	return
+}
+
+// newTestKeyStore returns a key store to use in tests
+func newTestKeyStore(t *testing.T, dir string) (keyStore *keystore.KeyStore, keyStoreDir string) {
+	var err error
+	keyStoreDir, err = ioutil.TempDir(os.TempDir(), dir)
+	require.NoError(t, err)
+	keyStore = keystore.NewKeyStore(keyStoreDir, keystore.LightScryptN, keystore.LightScryptP)
+	return
+}
+
+func TestCreateAndRecoverAccountSuccess(t *testing.T) {
+	accManager, nodeManager := newTestAccManager(t)
+
+	var password string
+
+	keyStore, keyStoreDir := newTestKeyStore(t, "accounts")
+	defer os.RemoveAll(keyStoreDir) //nolint: errcheck
+
+	// Don't fail on empty password
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	_, _, _, err := accManager.CreateAccount(password)
+	require.NoError(t, err)
+
+	password = "some-pass"
```

Comment:
For multiple test cases, I recommend using table-driven tests. You can read more about this approach here: https://blog.golang.org/subtests.

Here we have only two cases so it's not a big deal.

Reply:
Nice tip, thanks! I've actually seen that practice in `TestVerifyAccountPassword`.

However, I had to pair a successful creation and a recovery in `TestCreateAndRecoverAccountSuccess` since recovery uses both the input and one of the outputs of creation. And `TestCreateAndRecoverFail_KeyStore` is different from the first one regarding that. So it doesn't seem useful in this case I suppose.